stdlib/2and3/zipfile.pyi

```diff
@@ -0,0 +1,94 @@
+# Stubs for zipfile (Python 3.4)
+
+from typing import Callable, IO, Optional, Tuple, Type, Union
+from types import TracebackType
+import sys
+
+
+_SZI = Union[str, ZipInfo]
+_DT = Tuple[int, int, int, int, int, int]
+
+
+class BadZipFile(Exception): ...
+
+if sys.version_info > (3,):
+    BadZipfile = ...  # type: Type[BadZipFile]
+
+class LargeZipFile(Exception): ...
+
+class ZipFile:
+    debug = ...  # type: int
+    comment = ...  # type: bytes
+    def __init__(self, file: Union[str, IO[bytes]], mode: str = ..., compression: int = ...,
+                 allowZip64: bool = ...) -> None: ...
+    def __enter__(self) -> ZipFile: ...
+    def __exit__(self, exc_type: Optional[Type[BaseException]],
+                 exc_val: Optional[Exception],
+                 exc_tb: Optional[TracebackType]) -> bool: ...
+    def close(self) -> None: ...
+    def getinfo(self, name: str) -> None: ...
+    def infolist(self) -> List[ZipInfo]: ...
+    def namelist(self) -> List[str]: ...
+    def open(self, name: _SZI, mode: str = ...,
+             pwd: Optional[bytes] = ...) -> IO[bytes]: ...
+    def extract(self, member: _SZI, path: Optional[_SZI] = ...,
+                pwd: bytes = ...) -> str: ...
+    def extractall(self, path: Optional[str] = ...,
+                   members: Optional[List[str]] = ...,
+                   pwd: Optional[bytes] = ...) -> None: ...
+    def printdir(self) -> None: ...
+    def setpassword(self, pwd: bytes) -> None: ...
+    def read(self, name: _SZI, pwd: Optional[bytes] = ...) -> bytes: ...
+    def testzip(self) -> Optional[str]: ...
+    def write(self, filename: str, arcname: Optional[str] = ...,
+              compress_type: Optional[int] = ...) -> None: ...
+    if sys.version_info >= (3,):
+        def writestr(self, zinfo_or_arcname: _SZI, data: Union[bytes, str],
+                     compress_type: Optional[int] = ...) -> None: ...
+    else:
+        def writestr(self,  # type: ignore
```

Comment:
Sounds good. Is there a way to avoid the type:ignore though? Or maybe keep
it but add a comment saying why it's there.
On Jul 12, 2016 1:04 PM, "Matthias Kramm" <notifications@github.com> wrote:

> In stdlib/2and3/zipfile.pyi
> <https://github.com/python/typeshed/pull/345#discussion_r70478322>:
>
> > +    def extract(self, member: _SZI, path: Optional[_SZI] = ...,
> > +                pwd: bytes = ...) -> str: ...
> > +    def extractall(self, path: Optional[str] = ...,
> > +                   members: Optional[List[str]] = ...,
> > +                   pwd: Optional[bytes] = ...) -> None: ...
> > +    def printdir(self) -> None: ...
> > +    def setpassword(self, pwd: bytes) -> None: ...
> > +    def read(self, name: _SZI, pwd: Optional[bytes] = ...) -> bytes: ...
> > +    def testzip(self) -> Optional[str]: ...
> > +    def write(self, filename: str, arcname: Optional[str] = ...,
> > +              compress_type: Optional[int] = ...) -> None: ...
> > +    if sys.version_info >= (3,):
> > +        def writestr(self, zinfo_or_arcname: _SZI, data: Union[bytes, str],
> > +                     compress_type: Optional[int] = ...) -> None: ...
> > +    else:
> > +        def writestr(self,  # type: ignore
>
> There's nothing wrong with sys.version_info. As long as it's not
> literally every line, it's fine to have a combined stub.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/python/typeshed/pull/345/files/b1f4b474a9bc15fbb34a31ce86f3d8b48b1c9288#r70478322>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABCCAudJkkQA0hXsX0n-Hgi_3dD6f4TYks5qU8j9gaJpZM4JGG_v>
> .
>